apps/hexagon_benchmarks/process.cpp

```diff
@@ -0,0 +1,109 @@
+#include <stdio.h>
+#include <memory.h>
+#include <assert.h>
+#include <stdlib.h>
+
+#include "../support/benchmark.h"
+#include "process.h"
+
+void usage(char *prg_name) {
+    const char usage_string[] = " Run a bunch of small filters\n\n"
+                                "\t -m -> hvx_mode - options are hvx64, hvx128. Default is to run hvx64, hvx128 and cpu\n"
+                                "\t -n -> number of iterations\n"
+                                "\t -h -> print this help message\n";
+    printf ("%s - %s", prg_name, usage_string);
+
+}
+
+const char *to_string(bmark_run_mode_t mode) {
+    if (mode == bmark_run_mode_t::hvx64) {
+        return "(64 byte mode)";
+    } else if (mode == bmark_run_mode_t::hvx128) {
+        return "(128 byte mode)";
+    } else {
+        return "(cpu)";
+    }
+}
+
+int main(int argc, char **argv) {
+    // Set some defaults first.
+    const int W = 1024;
+    const int H = 1024;
+    std::vector<bmark_run_mode_t> modes = {bmark_run_mode_t::hvx64, bmark_run_mode_t::hvx128, bmark_run_mode_t::cpu};
+    int iterations = 10;
+
+    // Process command line args.
+    for (int i = 1; i < argc; ++i) {
+        if (argv[i][0] == '-') {
+            switch (argv[i][1]) {
+            case 'm':
+                {
+                    std::string mode_to_run = argv[i+1];
+                    modes.clear();
```

Comment:
Yes, thanks for spotting this. I had handled only the "-m args" case and not the "-m args -m arg2". Kinda like -O optimizations - only the last one is taken. I'll fix this.